### PR TITLE
Yared interlink rating with article

### PR DESCRIPTION
**Pull request by Yared-betsega**
Finished interlinking articles with ratings

### CODE REVIEW

File: starter-project-backend-g31/src/controllers/article.ts [hunk 1]
```diff
@@ -64,3 +63,16 @@ export const deleteOne = async (req: Request, res: Response) => {
     res.send(article);
 }
 
+export const addRatingToArticle = async (id: string, val: number) => {
+    const article = await Article.findById(id);
+    if (article != null) {
+        article.addRating(val)
```
Comment (RiyadHassen) [inline starter-project-backend-g31/src/controllers/article.ts:69]: better to add validation for val

Comment (RiyadHassen) [inline starter-project-backend-g31/src/controllers/article.ts:69]: better to add validation for val here

File: starter-project-backend-g31/src/controllers/rating.ts [hunk 2]
```diff
@@ -1,4 +1,6 @@
 import { Rating } from "../models/rating";
+import {Article} from '../models/article';
+import {addRatingToArticle, updateRatingForArticle} from "./article"
 
 export const getAllRatings = (req: any, res: any) => {
```
Comment (RiyadHassen) [inline starter-project-backend-g31/src/controllers/rating.ts:5]: we may not need this part alone, getAllRating for  a given article makes sense but without article it does not have functionality.

Comment (RiyadHassen) [inline starter-project-backend-g31/src/controllers/rating.ts:5]: we may not need this part alone, getAllRating for  a given aritcle makes sense but without article it does not have functionality.

**Review by RiyadHassen — CHANGES_REQUESTED** [review]

File: starter-project-backend-g31/src/controllers/article.ts (hunk 1 above)
Comment (Yared-betsega) [reply starter-project-backend-g31/src/controllers/article.ts:69]: done!

Comment (Yared-betsega) [reply starter-project-backend-g31/src/controllers/article.ts:69]: done!

File: starter-project-backend-g31/src/controllers/rating.ts (hunk 2 above)
Comment (Yared-betsega) [reply starter-project-backend-g31/src/controllers/rating.ts:4]: Yeah I removed the import for Article. but the second two are for interlinking purposes.

Comment (Yared-betsega) [reply starter-project-backend-g31/src/controllers/rating.ts:4]: Yeah I removed the import for Article. but the second two are for interlinking purposes.

**Review by RiyadHassen — APPROVED** [review]
LGTM

**Pull request merged**